plugins/ComposeAuth/src/commonMain/kotlin/io/github/jan/supabase/compose/auth/ComposeAuth.kt

```diff
@@ -0,0 +1,70 @@
+package io.github.jan.supabase.compose.auth
+
+import io.github.jan.supabase.SupabaseClient
+import io.github.jan.supabase.gotrue.LogoutScope
+import io.github.jan.supabase.gotrue.gotrue
+import io.github.jan.supabase.gotrue.providers.Google
+import io.github.jan.supabase.gotrue.providers.builtin.IDToken
+import io.github.jan.supabase.plugins.SupabasePlugin
+import io.github.jan.supabase.plugins.SupabasePluginProvider
+
+
+sealed interface ComposeAuth : SupabasePlugin {
+
+    data class Config(
+        var loginConfig: LoginConfig? = null
+    ) : SupabasePlugin
+
+    companion object : SupabasePluginProvider<Config, ComposeAuth> {
+
+        override val key: String = "composeauth"
+
+        const val API_VERSION = 1
+
+        override fun create(supabaseClient: SupabaseClient, config: Config): ComposeAuth {
+            return ComposeAuthImpl(config, supabaseClient)
+        }
+
+        override fun createConfig(init: Config.() -> Unit): Config {
+            return Config().apply(init)
+        }
+    }
+}
+
+class ComposeAuthImpl(
+    val config: ComposeAuth.Config,
+    val supabaseClient: SupabaseClient,
```

Comment:
I think you can make the supabase client public, all other plugins also do it and it looks a bit better when there is no `this as ComposeAuthImpl`